packages/react-native-analytics/src/integrations/castle/__tests__/Castle.test.js

```diff
@@ -0,0 +1,274 @@
+import CastleReactNativeIntegration, {
+  LOGGER_MESSAGE_PREFIX,
+  HTTP_CLIENT_ERROR_MESSAGE,
+} from '../';
+import Integration from '../../integration';
+import CastleIO from '@castleio/react-native-castle';
+import coreClient from '@farfetch/blackout-core/helpers/client';
+import utils from '@farfetch/blackout-core/analytics/utils';
+
+const mockRequestHeaderName = 'castle-header-foo';
+const mockRequestHeaderValue = '12342342345241342423424';
+jest.mock('@castleio/react-native-castle', () => {
+  return {
+    ...jest.requireActual('@castleio/react-native-castle'),
+    configure: jest.fn(() => Promise.resolve()),
+    requestTokenHeaderName: async () => Promise.resolve(mockRequestHeaderName),
+    createRequestToken: async () => Promise.resolve(mockRequestHeaderValue),
+    identify: jest.fn(() => Promise.resolve()),
+    reset: jest.fn(() => Promise.resolve()),
+  };
+});
+
+jest.mock('@farfetch/blackout-core/analytics/utils', () => ({
+  ...jest.requireActual('@farfetch/blackout-core/analytics/utils'),
+  logger: {
+    error: jest.fn(),
+    warn: jest.fn(),
+  },
+}));
+
+const mockIntegrationOptions = {
+  httpClient: {
+    defaults: {
+      headers: {
+        common: {},
+      },
+    },
+  },
+  configureHttpClientHeader: jest.fn(),
+};
+const getIntegrationInstance = async customOptions => {
+  return await CastleReactNativeIntegration.createInstance(
```

Comment:
The await here is unnecessary because the `createInstance` method will not return a promise. If you need to await the `initialize` method to finish, you can await the initializePromise on the instance directly like this:

```js
const instance = CastleReactNativeIntegration.createInstance(customOptions);

await instance.initializePromise();
```

Reply:
fixed 🙂  thanks!